src/posix/net.cpp

```diff
@@ -35,13 +35,19 @@ MUTEX_CLASS CurlHttpIO::curlMutex(false);
 #if defined(USE_OPENSSL) && !defined(OPENSSL_IS_BORINGSSL)
 
 MUTEX_CLASS **CurlHttpIO::sslMutexes = NULL;
+static MUTEX_CLASS lock_init_mutex;
 void CurlHttpIO::locking_function(int mode, int lockNumber, const char *, int)
 {
     MUTEX_CLASS *mutex = sslMutexes[lockNumber];
     if (mutex == NULL)
     {
-        mutex = new MUTEX_CLASS(true);
-        sslMutexes[lockNumber] = mutex;
+        // we still have to be careful about multiple threads getting to this point simultaneously
+        lock_init_mutex.lock();
+        while (!(mutex = sslMutexes[lockNumber]))
```

Comment:
I understand that the double check is needed because another thread could have created the mutex a thread passes the `if` above but is there any reason to have a `while` here instead of an `if`?

Reply:
the `while` was just to get that `mutex = sslMutexes[lockNumber]` executed a second time after the mutex was constructed and added to the array, so that we can use the value in the code below as usual.  But you're right, it's clearer for the reader to use an `if`, especially as we can easily do the assignment in the existing line.  Adjusted - cheers

Reply:
Great, thank you!